examples/viewer.py

```diff
@@ -0,0 +1,1489 @@
+        self.cfg.agents[self.agent_id] = self.default_agent_config()
+
+        if self.enable_batch_renderer:
+            self.cfg.enable_batch_renderer = True
+            self.cfg.sim_cfg.create_renderer = False
+            self.cfg.sim_cfg.enable_gfx_replay_save = True
+
+        if self.sim_settings["use_default_lighting"]:
+            logger.info("Setting default lighting override for scene.")
+            self.cfg.sim_cfg.override_scene_light_defaults = True
+            self.cfg.sim_cfg.scene_light_setup = (
+                habitat_sim.gfx.DEFAULT_LIGHTING_KEY
+            )
+
+        if self.sim is None:
+            self.tiled_sims = []
+            for _i in range(self.num_env):
+                self.tiled_sims.append(habitat_sim.Simulator(self.cfg))
+            self.sim = self.tiled_sims[0]
+        else:  # edge case
+            for i in range(self.num_env):
+                if (
+                    self.tiled_sims[i].config.sim_cfg.scene_id
+                    == self.cfg.sim_cfg.scene_id
+                ):
+                    # we need to force a reset, so change the internal config scene name
+                    self.tiled_sims[i].config.sim_cfg.scene_id = "NONE"
+                self.tiled_sims[i].reconfigure(self.cfg)
+
+        # post reconfigure
+        self.default_agent = self.sim.get_agent(self.agent_id)
+        self.render_camera = (
+            self.default_agent.scene_node.node_sensor_suite.get("color_sensor")
+        )
+
+        # set sim_settings scene name as actual loaded scene
+        self.sim_settings["scene"] = self.sim.curr_scene_name
+
+        # Initialize replay renderer
+        if self.enable_batch_renderer and self.replay_renderer is None:
+            self.replay_renderer_cfg = ReplayRendererConfiguration()
+            self.replay_renderer_cfg.num_environments = self.num_env
+            self.replay_renderer_cfg.standalone = (
+                False  # Context is owned by the GLFW window
+            )
+            self.replay_renderer_cfg.sensor_specifications = self.cfg.agents[
+                self.agent_id
+            ].sensor_specifications
+            self.replay_renderer_cfg.gpu_device_id = (
+                self.cfg.sim_cfg.gpu_device_id
+            )
+            self.replay_renderer_cfg.force_separate_semantic_scene_graph = (
+                False
+            )
+            self.replay_renderer_cfg.leave_context_with_background_renderer = (
+                False
+            )
+            self.replay_renderer = ReplayRenderer.create_batch_replay_renderer(
+                self.replay_renderer_cfg
+            )
+            # Pre-load composite files
+            if sim_settings["composite_files"] is not None:
+                for composite_file in sim_settings["composite_files"]:
+                    self.replay_renderer.preload_file(composite_file)
+
+        self.krm = KinematicRelationshipManager(self.sim)
+        self.krm.initialize_relationship_graph()
```

Comment:
If I understand correctly, we simply have to create a new instance upon changing episode. Is there anything else to do?

Reply:
TL;DR We should implement a specific RearrangeEpisode initializer for KRM where we add relationships between each added object and parent receptacle.

The current one is a fun tool for dynamic scenes, but we don't intend to use dynamics to settle the episodes, so this method won't work out of the box.